### PR TITLE
Sakomws build l2ami engops

### PR DESCRIPTION
**Pull request by sakomws**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+// Load the OG Jenkins Library
+@Library('OGJenkinsLib@3.6.1') _
+
+final GIT_REPOSITORY_NAME = 'amazon-eks-ami'
+final KUBERNETES_VERSION = '1.12.7'
+final PACKER_IMAGE_MANIFEST = 'manifest.json'
+def config = [:]  // Pipeline configuration
+
+def containers = [
+  OGContainer('devops', "${env.INTERNAL_REGISTRY_HOSTNAME}/devops", 'PR-18-2.3.0', [resourceLimitCpu: '2000m', resourceLimitMemory: '2Gi'])
```
Comment (Chili-Man) [inline Jenkinsfile]: 2.3.0

**Review by Chili-Man — CHANGES_REQUESTED** [review]

File: Jenkinsfile.properties [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    disableConcurrentBuilds() // Only allow this pipeline job to be ran once
+  ]
+)
+
+// Default if Account ID is not provided
+def awsAccountsOnMerge = ['development-aws', 'integration-aws', 'staging-aws', 'production-aws' , 'engops-aws']
+def awsAccountsOnPullRequest = ['development-aws']
+
+return [
+  regions: params['REGIONS'].split(','),
```
Comment (Chili-Man) [inline Jenkinsfile.properties]:
```suggestion
  regions: params['REGIONS'].split(',') as List,
```

File: Jenkinsfile.properties [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env groovy
+
+// Loads the Jenkins pipeline properties configuration
+def pipelineParameters = parameters(
+  [
+    booleanParam(name: 'DRY', defaultValue: true, description: 'test run'),
+    stringParam(name: 'REGIONS', defaultValue: 'us-west-2,us-east-1', description: 'comma separated list of aws regions'),
+    stringParam(name: 'AWS_ACCOUNT_CREDENTIAL_IDS', defaultValue: '', description: 'Comma separated Jenkins credential IDs for AWS accounts. Acceptable values are: development-aws,integration-aws,staging-aws,production-aws,engops-aws')
```
Comment (Chili-Man) [inline Jenkinsfile.properties:9]: mention that if its empty, then defaults will be used depending on if its a merge or a pull request

File: install-worker.sh [hunk 4]
```diff
@@ -168,8 +170,10 @@ BINARIES=(
 for binary in ${BINARIES[*]} ; do
     if [[ ! -z "$AWS_ACCESS_KEY_ID" ]]; then
         echo "AWS cli present - using it to copy binaries from s3."
+        set -x
```
Comment (Chili-Man) [inline install-worker.sh]: remove

File: install-worker.sh [hunk 5]
```diff
@@ -168,8 +170,10 @@ BINARIES=(
 for binary in ${BINARIES[*]} ; do
     if [[ ! -z "$AWS_ACCESS_KEY_ID" ]]; then
         echo "AWS cli present - using it to copy binaries from s3."
+        set -x 
         aws s3 cp --region $BINARY_BUCKET_REGION $S3_PATH/$binary .
         aws s3 cp --region $BINARY_BUCKET_REGION $S3_PATH/$binary.sha256 .
+        set +x
```
Comment (Chili-Man) [inline install-worker.sh]: remove

File: Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+              }
+            }
+
+          [accountId, job]
+        }
+
+        parallel jobs
+
+        archiveArtifacts(artifacts: PACKER_IMAGE_MANIFEST, fingerprint: true)
+
```
Comment (Chili-Man) [inline Jenkinsfile]: remove line

File: Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    container('devops') { 
+        def jobs = config.accountIds.collectEntries { accountId ->
+          def job = {
+              withCredentials([usernamePassword(credentialsId: accountId, passwordVariable: 'AWS_SECRET_ACCESS_KEY', usernameVariable: 'AWS_ACCESS_KEY_ID')]) {
+                if (config.dry) {
+                  echo "Would have ran: 'make VERSION=${KUBERNETES_VERSION} k8s'"
+                  sh "touch ${PACKER_IMAGE_MANIFEST}"
+                  
+                } else {
+                  sh "make VERSION=${KUBERNETES_VERSION} k8s"
```
Comment (Chili-Man) [inline Jenkinsfile]: only wrap this statement in the container('devops')

**Review by Chili-Man — CHANGES_REQUESTED** [review]
Add a OG-CHANGELOG.md which specifies the changes we added to this

File: install-worker.sh (hunk 5 above)
Comment (sakomws) [reply install-worker.sh]: ok

File: install-worker.sh (hunk 4 above)
Comment (sakomws) [reply install-worker.sh]: ok

File: Jenkinsfile.properties (hunk 2 above)
Comment (sakomws) [reply Jenkinsfile.properties]: ok

File: Jenkinsfile.properties (hunk 3 above)
Comment (sakomws) [reply Jenkinsfile.properties:9]: ok

**Review by Chili-Man — DISMISSED** [review]

**Review by Chili-Man — CHANGES_REQUESTED** [review]
Tagging did not working

File: Jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+  stage('Bake Encrypted AMIs') {
+    def jobs = config.accountIds.collectEntries { accountId ->
+      def job = {
+          withCredentials([usernamePassword(credentialsId: accountId, passwordVariable: 'AWS_SECRET_ACCESS_KEY', usernameVariable: 'AWS_ACCESS_KEY_ID')]) {
+            if (config.dry) {
+              echo "Would have ran: 'make VERSION=${KUBERNETES_VERSION} k8s'"
+              sh "touch ${PACKER_IMAGE_MANIFEST}"    
+            } else {
+               container('devops') { 
+                  sh "make VERSION=${KUBERNETES_VERSION} k8s"
```
Comment (Chili-Man) [inline Jenkinsfile]:
```suggestion
                 withEnv() {
                  sh "make VERSION=${KUBERNETES_VERSION} k8s"
                  }
```

File: Makefile [hunk 9]
```diff
@@ -60,6 +62,8 @@ k8s: validate
 		-var cni_version=$(CNI_VERSION) \
 		-var cni_plugin_version=$(CNI_PLUGIN_VERSION) \
 		-var docker_version=$(DOCKER_VERSION) \
+		-var og_image_version=$(OG_IMAGE_VERSION) \
+		-var ami_regions=$(ami_regions) \
```
Comment (Chili-Man) [inline Makefile]: capitalize

File: Jenkinsfile.properties [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    disableConcurrentBuilds() // Only allow this pipeline job to be ran once
+  ]
+)
+
+// Default if Account ID is not provided
+def awsAccountsOnMerge = ['development-aws', 'integration-aws', 'staging-aws', 'production-aws' , 'engops-aws']
+def awsAccountsOnPullRequest = ['development-aws']
+
+return [
+  regions: params['REGIONS'].split(',') as List,
```
Comment (Chili-Man) [inline Jenkinsfile.properties]: dont
```suggestion
  regions: params['REGIONS'],
```

**Review by Chili-Man — CHANGES_REQUESTED** [review]

File: Jenkinsfile [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+  stage('Bake Encrypted AMIs') {
+    def jobs = config.accountIds.collectEntries { accountId ->
+      def job = {
+          withCredentials([usernamePassword(credentialsId: accountId, passwordVariable: 'AWS_SECRET_ACCESS_KEY', usernameVariable: 'AWS_ACCESS_KEY_ID')]) {
+            if (config.dry) {
+              echo "Would have ran: 'make VERSION=${KUBERNETES_VERSION} k8s'"
+              sh "touch ${PACKER_IMAGE_MANIFEST}"    
+            } else {
+               container('devops') { 
+                  withEnv(["REGIONS=${AMI_REGIONS}"]) {
```
Comment (Chili-Man) [inline Jenkinsfile]:
```suggestion
                  withEnv(["AMI_REGIONS=${config.regions}"]) {
```

File: Jenkinsfile.properties [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    disableConcurrentBuilds() // Only allow this pipeline job to be ran once
+  ]
+)
+
+// Default if Account ID is not provided
+def awsAccountsOnMerge = ['development-aws', 'integration-aws', 'staging-aws', 'production-aws' , 'engops-aws']
+def awsAccountsOnPullRequest = ['development-aws']
+
+return [
+  regions: params['REGIONS'].split(','),
```
Comment (Chili-Man) [inline Jenkinsfile.properties]:
```suggestion
  regions: params['REGIONS'],
```

**Review by Chili-Man — CHANGES_REQUESTED** [review]

**Review by Chili-Man — APPROVED** [review]

File: Jenkinsfile [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    // Save the configuration as an artifact
+    utils.createArtifact('initialConfig', config)
+  }
+
+  stage('Bake Encrypted AMIs') {
+    def jobs = config.accountIds.collectEntries { accountId ->
+      def job = {
+        withCredentials([usernamePassword(credentialsId: accountId, passwordVariable: 'AWS_SECRET_ACCESS_KEY', usernameVariable: 'AWS_ACCESS_KEY_ID')]) {
+          if (config.dry) {
+            echo "Would have ran: 'make VERSION=${KUBERNETES_VERSION} k8s'"
```
Comment (jspiro) [inline Jenkinsfile:50]: Declaring that I find this line BS and am approving under duress.

**Review by jspiro — APPROVED** [review]

**Pull request merged**